lib/Core/Executor.cpp

```diff
@@ -4697,8 +4703,45 @@ void Executor::run(ExecutionState *initialState) {
 
   objectManager->initialUpdate();
 
+  auto lastExecutionTime = std::chrono::steady_clock::now();
+  Delta dt;
+  std::thread consumer;
+  std::thread producer;
+  StatisticQueue StatQ;
+
+  dt.initPrevDelta(objectManager->StatisticMap);
+  ServerConnection sc;
+  sc.url = "http://localhost:8080/metrics";
+  sc.getUID();
+  std::atomic<bool> shouldStop = false;
+  if (sc.UID != "ServerNotValid") {
+    consumer = std::thread([&StatQ, &dt, &sc, &shouldStop]() {
+      while (!shouldStop) {
+        if (!StatQ.empty()) {
+          auto data = StatQ.pop();
+          sc.PostRequest(dt.SerializeDelMap(data, sc.UID));
```

Comment:
As I see, delta stores the last map which was sent to the server.
However, I would not expect that from the object named `Delta`.

In the perfect world serialising function should be marked as const and do not modify anything in the Delta.

Reply:
Additionally, pass data by rvalue.